### PR TITLE
Add source build switches in order to be able to build source-build on windows

### PR DESCRIPTION
**Pull request by safern**
I'm not sure if this should go against master.

The windows build script was missing the DotNetBuildFromSource and DotNetCoreSdkDir parameters in order to build msbuild on source-build as we do for non-Windows.

cc: @rainersigwald @weshaggard

### CODE REVIEW

File: build/build.ps1 [hunk 1]
```diff
@@ -210,9 +218,18 @@ function Build {
 
   echo "Repo toolset used from: $RepoToolsetBuildProj"
 
-  $solution = Join-Path $RepoRoot "MSBuild.sln"
+  $createTlb = $true
```
Comment (rainersigwald) [inline build/build.ps1]: Don't explicitly set this to `true`; that'll try to create TLBs for projects that don't need them. Set it to false or leave it unset.

Comment (safern) [reply build/build.ps1]: Ok, sounds good.

**Comment by cdmihai** [comment]
Why the need to support sourcebuild on Windows? Wasn't it needed only for Redhat?

File: build/build.ps1 [hunk 2]
```diff
@@ -210,9 +218,18 @@ function Build {
...
+    $solution = Join-Path $RepoRoot "MSBuild.SourceBuild.sln"
+    $createTlb = $false
+  }
+  else
+  {
+    $solution = Join-Path $RepoRoot "MSBuild.sln"
+  }
 
-  $commonMSBuildArgs = "/m", "/clp:Summary", "/v:$verbosity", "/p:Configuration=$configuration", "/p:Projects=$solution", "/p:CIBuild=$ci", "/p:RepoRoot=$reporoot"
+  $commonMSBuildArgs = "/m", "/clp:Summary", "/v:$verbosity", "/p:Configuration=$configuration", "/p:Projects=$solution", "/p:CIBuild=$ci", "/p:RepoRoot=$reporoot", "/p:DisableNerdbankVersioning=$DotNetBuildFromSource", "/p:CreateTlb=$createTlb"
```
Comment (cdmihai) [inline build/build.ps1]: The NerdbankVersioning restriction comes from Redhat. Is this still valid for Windows?

**Comment by safern** [comment]
> Why the need to support sourcebuild on Windows? Wasn't it needed only for Redhat?
 
We want our developers to be able to produce an SDK or a runtime with their changes using source build infrastructure to simplify this process instead of having them to build all the repos on the stack.

**Comment by weshaggard** [comment]
>Why the need to support sourcebuild on Windows? Wasn't it needed only for Redhat

Redhat is one of the first targets but not only one. We need to make the source-build work on all the OS's. For our devs working on the full product on windows it is still very much interesting to build the entire stack on windows.

File: build/build.ps1 (hunk 2 above)
Comment (safern) [reply build/build.ps1]: I've deleted this as we're already passing this msbuild parameter through source-build: 

https://github.com/dotnet/source-build/blob/dev/release/2.1/repos/msbuild.proj#L9

File: build/build.ps1 (hunk 1 above)
Comment (safern) [reply build/build.ps1]: Fixed.

**Pull request merged**